### PR TITLE
Lua Network API

### PR DESCRIPTION
**Pull request by madmaxoft**
Exported the cNetwork API to Lua, so that plugins can use it.

### CODE REVIEW

File: src/Bindings/LuaState.h [hunk 1]
```diff
@@ -355,7 +366,7 @@ class cLuaState
 
 	/** Variadic template: Counting args means add one to the count of the rest. */
 	template <typename T, typename... Args>
-	int CountArgs(T, Args... args)
+	int CountArgs(T &, Args &&... args)
 	{
 		return 1 + CountArgs(args...);
 	}
```
Comment (worktycho) [inline src/Bindings/LuaState.h]: I should really replace this method with the `sizeof...` operator.

Comment (madmaxoft) [inline src/Bindings/LuaState.h]: I had no idea that thing existed. It could be the easiest fix.

Comment (worktycho) [inline src/Bindings/LuaState.h]: I'm building the change now.

**Comment by madmaxoft** [comment]
I'll rebase this onto current master to make it auto-mergable again.

File: MCServer/Plugins/NetworkTest/NetworkTest.lua [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+			end,  -- OnAccepted()
+			
+			-- There was an error listening on the port:
+			OnError = function (a_ErrorCode, a_ErrorMsg)
+				LOGINFO("FortuneServer(" .. a_Port .. ": Cannot listen: " .. a_ErrorCode .. " (" .. a_ErrorMsg .. ")")
+			end,  -- OnError()
+		}  -- Listen callbacks
+	end,  -- fortune
+	
+	-- TODO: Other services (fortune, daytime, ...)
```
Comment (NiLSPACE) [inline MCServer/Plugins/NetworkTest/NetworkTest.lua]: You say other services like `fortune`, but it's defined right above here right?

Comment (madmaxoft) [inline MCServer/Plugins/NetworkTest/NetworkTest.lua]: yea, forgot to update the comment.

**Comment by madmaxoft** [comment]
I think this might be ready for a merge

**Comment by Howaner** [comment]
:+1:

**Pull request merged**